### DISCUSSION TITLE
Can i use jsdoc @param in my documentation for controls

### DISCUSSION THREAD

**Discussion by SJongbloets** [storybookjs/storybook#28830] [Help]
### Summary

We're currently looking into adding more documentation to our storybook as we're building a small component library. 
The JSDoc is getting picked up by storybook and I can see this on the components page. However currently the @param values are not being picked up and added to the properties description field. 

![image](https://github.com/user-attachments/assets/bfa28e0b-b394-476c-ae92-c99f19a9e305)

For other components we have managed to do it, by commenting the types themselves, but i'd prefer to have my documentation at one location so we don't need to do things twice.

### Additional information

Storybook version: 8.2.8

```
Storybook plugins
    "@storybook/addon-a11y": "^8.2.8",
    "@storybook/addon-actions": "^8.2.8",
    "@storybook/addon-controls": "^8.2.8",
    "@storybook/addon-docs": "^8.2.8",
    "@storybook/addon-essentials": "^8.2.8",
    "@storybook/addon-interactions": "^8.2.8",
    "@storybook/addon-links": "^8.2.8",
    "@storybook/blocks": "^8.2.8",
    "@storybook/preset-scss": "^1.0.3",
    "@storybook/react": "^8.2.8",
    "@storybook/react-vite": "^8.2.8",
    "@storybook/test": "^8.2.8",
```

```typescript
// main.ts
import { dirname, join } from 'path';
import { StorybookConfig } from '@storybook/react-vite';

/**
 * This function is used to resolve the absolute path of a package.
 * It is needed in projects that use Yarn PnP or are set up within a monorepo.
 */
const getAbsolutePath = (value: string) => dirname(require.resolve(join(value, 'package.json')));

const config: StorybookConfig = {
  stories: ['../docs/**/*.@(mdx|stories.@(js|jsx|ts|tsx))'],

  addons: [
  
  
    getAbsolutePath('@storybook/addon-links'),
    getAbsolutePath('@storybook/addon-controls'),
    getAbsolutePath('@storybook/addon-essentials'),
    getAbsolutePath('@storybook/addon-interactions'),
    getAbsolutePath('@storybook/addon-a11y'),
  ],

  framework: '@storybook/react-vite',
  staticDirs: ['../stylesheets', 'public'],

  viteFinal: async (config) => {
    return config;
  },

  docs: {},

  typescript: {
    reactDocgen: 'react-docgen-typescript',
  },
};

export default config;
```

```typescript
// preview.ts
import { Preview } from '@storybook/react';

const preview: Preview = {
  parameters: {
    docs: {
      source: {
        state: 'open',
      },
    },
    actions: { argTypesRegex: '^on[A-Z].*' },
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/,
      },
    },
    options: {
      storySort: {
        order: ['Intro', ['Introduction', 'Custom Classes', 'CSS-variables'], 'Atoms', 'Molecules', 'Organisms'],
      },
    },
  },
  tags: ['autodocs'],
};

export default preview;

```

**Comment by cssinate** [comment]
Well, for starters, look at what your JSDoc is trying to say. It says your function accepts the parameters `name`, `draw`, `color`, etc. But your function _doesn't_ accept those parameters. It accepts one parameter, let's call it `iconProps`, and that parameter is an object with properties.

If your component file is in TS, then the better way to handle this is:

```ts
type IconProps = {
  /** The filename without the extension */
  name: string;
}

const Icon = ({ name, ...}: IconProps) => {
  ...
}
```

If your Icon file is _not_ using TS, then _maybe_ this will work?

```js
/**
 * @param {Object} iconProps
 * @param {string} iconProps.name - The filename without the extension
 */
```

This might work. I'm not sure.

**Reply by SJongbloets** [reply]
I've completely looked over the part where it deconstructed the props. I did try your suggestion but this sadly doesn't work either. 
For now i've just added the documentation to the prop types themselves to have the documentation in storybook

**Comment by duraki** [comment]
@cssinate while we are at this solution (btw. thanks for providing), is it possible to force MDX compiler to see line break as a new line, ie. by not providing two spaces at the end of the line?

For example, this would render everything in one line:

```jsx
/**
 * @component
 * @param {object} props - The properties for the component.
 * @param {string} [props.className] - Custom class names for styling.
```

But this, of course, will render it each in new lines, due to spaces at the end:

```jsx
/**
 * @component  
 * @param {object} props - The properties for the component.  
 * @param {string} [props.className] - Custom class names for styling.  
```

Can we somehow force Storybook MDX to break in new lines after literal line break was provided in Markdown?
